chainer/sequential.py

```diff
@@ -0,0 +1,393 @@
+        :class:`~Sequential` model easily with the same ways as Python lists::
+
+          model_A = Sequential(L.Linear(10, 10), F.relu)
+          model_B = Sequential(L.Linear(10, 10), F.sigmoid)
+          model_C = model_A + model_B
+
+        To repeat a :class:`~Sequential` object multiple times, you can use
+        :meth:`~chainer.Link.repeat` method.
+
+          model_D = model_A.repeat(3)
+
+        You can also add your own functions or any callable objects to a
+        :class:`~Sequential` object::
+
+          from link.Links.model.vision.vgg import VGG16Layers()
+
+          model = Sequential()
+          model.append(L.Linear(n_out, n_hidden))
+          model.append(F.relu)
+          model.append(F.Reshape((1, 3, 224, 224)))
+          model.append(VGG16Layers())
+          model.append(lambda x: x['prob'])
+
+          y = model(x)
+
+        The above code example shows how to add some layers to the ``model``
+        using :meth:`~Sequential.append` method and then add a large network
+        (``VGG16Layers``) and finally add a lambda function to extract the
+        ``prob`` output.
+
+        Note that a :class:`~Sequential` link which has at least one ``lambda``
+        function as its member cannot be pickled. So, please use ``partial``
+        method from ``functools`` package instead::
+
+          from functools import partial
+
+          # This is not pickable
+          model = Sequential(
+              L.Convolution2D(None, 64, 3, 1, 1),
+              lambda x: F.max_pooling_2d(x, 2)
+          )
+
+          # This is pickable
+          model = Sequential(
+              L.Convolution2D(None, 64, 3, 1, 1),
+              partial(F.max_pooling_2d, ksize=2)
+          )
+
+    Args:
+        layers: The layers which are called in its order. Each component should
+            be a callable object including :class:`~Link` object and
+            functions defined under the :mod:`chainer.functions`, e.g.,
+            :obj:`~chainer.functions.relu`, etc.
+
+    """
+
+    def __init__(self, *layers):
+        super(Sequential, self).__init__()
+        self._layers = []
+        for layer in layers:
+            self.append(layer)
+
+    def __len__(self):
+        return len(self._layers)
+
+    def __getitem__(self, i):
+        return self._layers[i]
+
+    def __setitem__(self, i, layer):
+        if i >= len(self):
+            raise ValueError(
+                '{} should be less than {}'.format(i, len(self)))
+        if not callable(layer):
+            raise ValueError(
+                'All elements of a Sequential class should be callable. But '
+                'given {} is not callable.'.format(layer))
+
+        if self._layers[i] is not layer:
+            del self[i]
+            self.insert(i, layer)
+
+    def __delitem__(self, i):
+        layer = self._layers.pop(i)
+        if isinstance(layer, link.Link):
+            for i, _link in enumerate(self._children):
```

Comment:
Is it correct to delete the link whose index is the smallest? Can we assume that the same link object appears only once in the sequential? (I think inserting the same link multiple times is not considered in even ChainList, so we may ignore such situation)

Reply:
Thank you for pointing out this. I fixed this part to use the name of `layer` to determine which element should be removed from `_children` list.